C/c4Database.cc

```diff
@@ -54,6 +54,13 @@ C4EncryptionKey C4EncryptionKeyFromPassword(slice password, C4EncryptionAlgorith
     return key;
 }
 
+std::array<uint8_t,32> C4EncryptionKeyFromPasswordSHA1(fleece::slice password) {
+    std::array<uint8_t,32> key{};
+    AssertParam(password.size > 0, "Password is empty");
+    if (!litecore::DeriveKeyFromPasswordSHA1(password, key.data(), key.size()))
+        C4Error::raise(LiteCoreDomain, kC4ErrorCrypto, "SHA1 Key derivation failed");
```

Comment:
In C4EncryptionKeyFromPassword, we also set algorithm to the key as well. Do we need to set anything here?